pkg/server/server.go

```diff
@@ -476,3 +571,223 @@ func adaptContext(parent genericapiserver.PostStartHookContext) context.Context
 	}(parent.StopCh)
 	return ctx
 }
+
+type kcpCRDLister struct {
+	crdLister       apiextensionslisters.CustomResourceDefinitionLister
+	workspaceLister tenancylisters.WorkspaceLister
+}
+
+var _ apiextensionslisters.CustomResourceDefinitionLister = (*kcpCRDLister)(nil)
+
+func (c *kcpCRDLister) List(selector labels.Selector) ([]*apiextensionsv1.CustomResourceDefinition, error) {
+	return c.crdLister.ListWithContext(context.Background(), selector)
+}
+
+func (c *kcpCRDLister) ListWithContext(ctx context.Context, selector labels.Selector) ([]*apiextensionsv1.CustomResourceDefinition, error) {
+	cluster, err := genericapirequest.ValidClusterFrom(ctx)
+	if err != nil {
+		return nil, err
+	}
+
+	// Check for API inheritance
+	inheriting := false
+	inheritFrom := ""
+	const adminClusterName = "admin" // FIXME
```

Comment:
You could look to see which lcluster the workspace CRD exists in

Reply:
Couldn't it exist in multiple? e.g. if we have multiple "org workspaces" == "lclusters that have the workspace apis"?

Reply:
would leave it as simple as this until we have a better idea about workspace location and identity